scipy/spatial/transform/tests/test_rotation_groups.py

```diff
@@ -151,3 +152,12 @@ def test_group_symmetry(name, size):
     distance = np.sort(cdist(q, q))
     deltas = np.max(distance, axis=0) - np.min(distance, axis=0)
     assert (deltas < TOL).all()
+
+
+@pytest.mark.parametrize("name", NAMES)
+def test_reduction(name):
```

Comment:
I think you can come up with stronger tests.

One test is to compare the result with straightforward triple loop implementation.

Reply:
I added a couple of tests and fixed a bug where the return value was wrong for
```
p.reduce(left=None, right=None, return_indices=True)
```